File: itcloud/src/test/java/com/itinfo/itcloud/service/ItDataCenterServiceTest.java
package com.itinfo.itcloud.service;

import com.itinfo.itcloud.model.computing.DataCenterVo;
import com.itinfo.itcloud.model.computing.EventVo;
import com.itinfo.itcloud.model.create.DataCenterCreateVo;
import com.itinfo.itcloud.model.error.CommonVo;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Order;
import org.junit.jupiter.api.Test;
import org.ovirt.engine.sdk4.types.DataCenter;
import org.ovirt.engine.sdk4.types.QuotaModeType;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
class ItDataCenterServiceTest {
    @Autowired ItDataCenterService dcService;

    String defaultDcId = "9c72ff12-a5f3-11ee-941d-00163e39cb43";

    @Test
    @DisplayName("데이터센터 리스트")
    void getList() {
        List<DataCenterVo> dcList = dcService.getList();

        assertThat(2).isEqualTo(dcList.size());
    }

    @Test
    @DisplayName("데이터센터 생성")
    void addDatacenter() {
        DataCenterCreateVo dc =
                DataCenterCreateVo.builder()
                .name("dfdf")
                .comment("testComment")
                .description("testDescription")
                .version("4.7")
                .storageType(false)
                .quotaMode(QuotaModeType.AUDIT)
                .build();

        CommonVo<Boolean> result =  dcService.addDatacenter(dc);
        assertThat(result.getHead().getCode()).isEqualTo(201);

        assertThat("dfdf").isEqualTo(dc.getName());
        assertThat("testComment").isEqualTo(dc.getComment());
        assertThat("testDescription").isEqualTo(dc.getDescription());
        assertThat("4.7").isEqualTo(dc.getVersion());
        assertThat(false).isEqualTo(dc.isStorageType());
        assertThat(QuotaModeType.AUDIT).isEqualTo(dc.getQuotaMode());
    }

    @Test
    @DisplayName("데이터센터 생성 - 이름 중복")
    void addDatacenter2() {
        DataCenterCreateVo dc =
                DataCenterCreateVo.builder()
                        .name("dfdf")
                        .comment("testComment")
                        .description("testDescription")
                        .version("4.7")
                        .storageType(false)
                        .quotaMode(QuotaModeType.AUDIT)
                        .build();

        CommonVo<Boolean> result =  dcService.addDatacenter(dc);
        assertThat(result.getHead().getCode()).isEqualTo(404);
    }

    @Test
    @DisplayName("데이터센터 수정 창")
    void getDatacenter() {
        String id = "5e3e972b-7000-4e8e-a43b-d90fd3e304a6";
        DataCenterCreateVo result =  dcService.setDatacenter(id);

        DataCenterCreateVo dc = DataCenterCreateVo.builder()
                .id(id)
                .name(result.getName())
                .comment(result.getComment())
                .description(result.getDescription())
                .version(result.getVersion())
                .storageType(result.isStorageType())
                .quotaMode(result.getQuotaMode())
                .build();

        assertThat(id).isEqualTo(result.getId());
        assertThat(dc.getName()).isEqualTo(result.getName());
        assertThat(result).isEqualTo(dc);

        System.out.println(dc);
    }

    @Test
    @DisplayName("데이터센터 수정")
    void editDatacenter() {
        String id = "5e3e972b-7000-4e8e-a43b-d90fd3e304a6";

        DataCenterCreateVo dc = DataCenterCreateVo.builder()
                .id(id)
                .name("tsfe")
                .comment("testedit")
                .description("s")
                .version("4.7")
                .storageType(false)
                .quotaMode(QuotaModeType.AUDIT)
                .build();

        CommonVo<Boolean> result = dcService.editDatacenter(id, dc);
        assertThat(result.getHead().getCode()).isEqualTo(201);
    }

    @Test
    @DisplayName("데이터센터 수정 - 이름 중복")
    void editDatacenter2() {
        String id = "5e3e972b-7000-4e8e-a43b-d90fd3e304a6";

        DataCenterCreateVo dc = DataCenterCreateVo.builder()
                .id(id)
                .name("Default")
                .comment("testComment")
                .description("testDescription")
                .version("4.7")
                .storageType(false)
                .quotaMode(QuotaModeType.AUDIT)
                .build();

        CommonVo<Boolean> result = dcService.editDatacenter(id, dc);
        assertThat(result.getHead().getCode()).isEqualTo(404);
    }


    @Test
    @DisplayName("데이터센터 삭제")
    void deleteDatacenter() {
        String id = "5e3e972b-7000-4e8e-a43b-d90fd3e304a6";

        CommonVo<Boolean> result = dcService.deleteDatacenter(id);
        assertThat(result.getHead().getCode()).isEqualTo(200);
    }

    @Test
    @DisplayName("데이터센터 이벤트 목록")
    void getEvent() {
        List<EventVo> result = dcService.getEvent(defaultDcId);
        assertThat(result.size()).isEqualTo(12);
    }
}

File: itcloud/src/test/java/com/itinfo/itcloud/service/ItDataCenterServiceTest.kt
package com.itinfo.itcloud.service

import com.itinfo.common.LoggerDelegate
import org.junit.jupiter.api.Test
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.hamcrest.Matchers.not
import org.hamcrest.Matchers.nullValue
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

/**
 * [ItDataCenterServiceTest]
 * [ItDataCenterService]에 대한 단위테스트
 *
 * @author chanhi2000
 * @since 2024.03.05
 */
@SpringBootTest
class ItDataCenterServiceTest {
	@Autowired
	private lateinit var service: ItDataCenterService

	/**
	 * [should_getName]
	 * [ItDataCenterService.getName]에 대한 단위테스트
	 * 
	 * @see ItDataCenterService.getName
	 **/
	@Test
	fun should_getName() {
		log.debug("should_getName ... ")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_getList]
	 * [ItDataCenterService.getList]에 대한 단위테스트
	 * 
	 * @see ItDataCenterService.getList
	 **/
	@Test
	fun should_getList() {
		log.debug("should_getList ... ")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_getEvent]
	 * [ItDataCenterService.getEvent]에 대한 단위테스트
	 * 
	 * @see ItDataCenterService.getEvent
	 **/
	@Test
	fun should_getEvent() {
		log.debug("should_getEvent ... ")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_getDatacenter]
	 * [ItDataCenterService.getDatacenter]에 대한 단위테스트
	 * 
	 * @see ItDataCenterService.getDatacenter
	 **/
	@Test
	fun should_getDatacenter() {
		log.debug("should_getDatacenter ... ")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_addDatacenter]
	 * [ItDataCenterService.addDatacenter]에 대한 단위테스트
	 * 
	 * @see ItDataCenterService.addDatacenter
	 **/
	@Test
	fun should_addDatacenter() {
		log.debug("should_addDatacenter ... ")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_editDatacenter]
	 * [ItDataCenterService.editDatacenter]에 대한 단위테스트
	 * 
	 * @see ItDataCenterService.editDatacenter
	 **/
	@Test
	fun should_editDatacenter() {
		log.debug("should_editDatacenter ... ")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}

	/**
	 * [should_deleteDatacenter]
	 * [ItDataCenterService.deleteDatacenter]에 대한 단위테스트
	 * 
	 * @see ItDataCenterService.deleteDatacenter
	 **/
	@Test
	fun should_deleteDatacenter() {
		log.debug("should_deleteDatacenter ... ")
		assertThat(service, `is`(not(nullValue())))
		// TODO: 메소드의 결과값에 대한 검증처리
	}


	companion object {
		private val log by LoggerDelegate()
	}
}